packages/goto/scripts/postinstall.js

```diff
@@ -35,25 +37,13 @@ const rulesFromURL = async url => {
   return body
 }
 
-const rulesFromURLs = async urls => {
-  const lists = await Promise.all(urls.map(rulesFromURL))
-
-  // Parse all lists
-  const { networkFilters, cosmeticFilters } = parseFilters(lists.join(EOL), {
-    debug: true
-  })
-
-  // Return cleaned version of the lists (no comments, no spaces, etc.)
-  return [
-    ...new Set([...networkFilters.map(f => f.rawLine), ...cosmeticFilters.map(f => f.rawLine)])
-  ]
-}
-
-const toTxt = (filepath, data) => writeFile(filepath, data)
-
 const main = async urls => {
-  const rules = await rulesFromURLs(urls)
-  await toTxt('src/rules.txt', rules.join(EOL))
+  const rules = Array.from(
+    getLinesWithFilters((await Promise.all(urls.map(rulesFromURL))).join(EOL))
+  )
+  const engine = PuppeteerBlocker.parse(rules.join(EOL))
+  engine.updateResources(await rulesFromURL(RESOURCES), '' + Date.now())
+  await writeFile('src/engine.bin', engine.serialize())
```

Comment:
are you serializing into a binary file for any reason? is it because of perf?

Reply:
This is mostly for performance reason yes. Initializing an instance of the adblocker using the `parse(...)` method might take up to 1-2 seconds if lots of filters are used (since it needs to do a lot of parsing and initialization of internal data-structures). But once all this work is done you can very efficiently convert the adblocker into a binary blob (i.e.: `Uint8Array`) and store it in a file. You can then restore the original, identical adblocker from this typed array *very* fast (around 1-10 ms on cold start). So basically calling `parse(...)` multiple times on the same filters looks like a good use-case of serialization instead, to perform the work once then just load the same engine from disk.